### DISCUSSION TITLE
Expert to link sqlite with Deno

### DISCUSSION THREAD

**Discussion by yolio2003** [prisma/prisma#21847] [Q&A]
### Bug description

follow the document to gen the edge.ts for deno, but still got error.

the error is 
```bash
...

Error validating datasource `db`: the URL must start with the protocol `prisma://`
    at Jn.handleRequestError (file:///Users/monkeytao/Documents/sub-tc/tcdeno/prisma/client/runtime/edge-esm.js:36:7159)
    at Jn.handleAndLogRequestError (file:///Users/monkeytao/Documents/sub-tc/tcdeno/prisma/client/runtime/edge-esm.js:36:6547)
    at Jn.request (file:///Users/monkeytao/Documents/sub-tc/tcdeno/prisma/client/runtime/edge-esm.js:36:6245)
    at async u (file:///Users/monkeytao/Documents/sub-tc/tcdeno/prisma/client/runtime/edge-esm.js:41:10327)
    at async main (file:///Users/monkeytao/Documents/sub-tc/tcdeno/prisma.ts:29:22) {
  name: "PrismaClientKnownRequestError",
  code: "P5002",
  clientVersion: "5.5.2",
  meta: undefined
}
```

### How to reproduce

```bash
deno run -A npm:prisma@5.5.2 init
deno run -A npm:prisma@5.5.2 db push
deno run -A --unstable npm:prisma@5.5.2 generate --data-proxy

deno run -A ./prisma.ts
```

### Expected behavior

i wish this just works

### Prisma information

```prisma

// This is your Prisma schema file,
// learn more about it in the docs: https://pris.ly/d/prisma-schema

generator client {
  provider = "prisma-client-js"
  previewFeatures = ["deno"]
  output = "./client"
}

datasource db {
  provider = "sqlite"
  url      = "file:./dev.db"
}

model Dinosaur {
  id          Int     @id @default(autoincrement())
  name        String  @unique
  description String
}
```

```ts
// Add your code using Prisma Client

import { PrismaClient } from './prisma/client/deno/edge.ts'

const prisma = new PrismaClient({
  datasources: {
    db: {
      url: 'file:./prisma/dev.db',
    },
  },
  log: [
    { level: 'warn', emit: 'event' },
    { level: 'info', emit: 'event' },
    { level: 'error', emit: 'event' },
  ],
})

prisma.$on('warn', (e) => {
  console.log(e)
})

prisma.$on('info', (e) => {
  console.log(e)
})

prisma.$on('error', (e) => {
  console.log(e)
})

async function main() {
  const countUsers = await prisma.dinosaur.count({})
}

main()
  .then(async () => {
    await prisma.$disconnect()
  })
  .catch(async (e) => {
    console.error(e)
    await prisma.$disconnect()
    Deno.exit(1)
  })

// deno run -A ./prisma.ts
```

### Environment & setup

- OS: Mac air m2
- Database: sqlite
- Node.js version: 19.0.0
- Deno version: 1.38.0

### Prisma Version

```
prisma                  : 5.5.2
@prisma/client          : 5.5.2
Current platform        : darwin-arm64
Query Engine (Node-API) : libquery-engine aebc046ce8b88ebbcb45efe31cbe7d06fd6abc0a (at node_modules/.pnpm/@prisma+engines@5.5.2/node_modules/@prisma/engines/libquery_engine-darwin-arm64.dylib.node)
Schema Engine           : schema-engine-cli aebc046ce8b88ebbcb45efe31cbe7d06fd6abc0a (at node_modules/.pnpm/@prisma+engines@5.5.2/node_modules/@prisma/engines/schema-engine-darwin-arm64)
Schema Wasm             : @prisma/prisma-schema-wasm 5.5.1-1.aebc046ce8b88ebbcb45efe31cbe7d06fd6abc0a
Default Engines Hash    : aebc046ce8b88ebbcb45efe31cbe7d06fd6abc0a
Studio                  : 0.494.0
Preview Features        : deno
```

**Comment by janpio** [comment]
Which documentation are you following?

**Reply by yolio2003** [reply]
https://docs.deno.com/runtime/tutorials/how_to_with_npm/prisma
this one

**Reply by janpio** [reply]
The tutorial you linked to does not work with SQLite, but PostgreSQL or some other online database.

Your error is caused by skipping this step:
https://docs.deno.com/runtime/tutorials/how_to_with_npm/prisma#setup-prisma-data-platform

Here are outdated instructions to create a Data Proxy project.
This has been replaced with Prisma Accelerate, so use https://www.prisma.io/data-platform/accelerate instead.
It will give you a `prisma://` connection string to use.

**Reply by yolio2003** [reply]
Thank you for your response. If I want to use SQLite with Prisma in Deno, are there any better resources you can recommend?

**Reply by janpio** [reply]
Not right now, I do not even know if this works.
You would definitely have to import `@prisma/client` (without the `/edge`) for this to work.

**Reply by yolio2003** [reply]
ok thx master i will check this later, infact i use a isolate service(nodejs project) to do this now.

**Reply by nicolaefilat** [reply]
Did you manage to fix this issue? 
I am running into the same error.

**Reply by yolio2003** [reply]
> Did you manage to fix this issue? I am running into the same error.

No, for some reason, I have switched to other solutions~

**Comment by isaced** [comment]
see https://github.com/denoland/deno/issues/27195

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
